src/main/java/de/zalando/aruha/nakadi/util/TimeBasedCache.java

```diff
@@ -0,0 +1,68 @@
+package de.zalando.aruha.nakadi.util;
+
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.function.Function;
+
+public class TimeBasedCache<Key, Value> {
```

Comment:
It's cool that you decided to build your own cache :) But wasn't there anything which is already created and suits our needs?
E.g. in google guava there is a CacheBuilder https://google.github.io/guava/releases/19.0/api/docs/com/google/common/cache/CacheBuilder.html